client/src/components/breadcrumbs.jsx

```diff
@@ -0,0 +1,39 @@
+//import { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import './Breadcrumbs.css'
+
+// internal logic for breadcrumbs:
+// "/"   -->  ["home"]
+// "/one" --> ["home", "one"]
+// "/one/one_one" --> ["home", "one", "one_one"] --> home / one / one_one
+//
+// hint:
+// ["home", "one", "one_one"].join("/")
+
+const Breadcrumbs = ({ currentPath, fetchFiles }) => {
+  const pathnames = currentPath === '/' ? ['home'] : ['home', ...currentPath.split('/').slice(1)]
+
+  // const pathnames = currentPath === '/' ? [] : currentPath.split('/').slice(1) I don't understand we should show the wotld HOME or no)
+
+  return (
+    <nav aria-label='Breadcrumb' className='breadcrumbs'>
+      <ol className='breadcrumb'>
+        {pathnames.map((name, index) => {
+          const path = index === 0 ? '/' : `/${pathnames.slice(1, index + 1).join('/')}`
+          return (
+            <li key={index} className='breadcrumb-item' onClick={() => fetchFiles(path)}>
+              {name}
+            </li>
+          )
+        })}
+      </ol>

```

Comment:
remove line 14 and change these highlighted lines here to something like this:
```jsx
<ul>
  <li onClick={() => blah("/")}>home</li>
  {
    // here is the "conditional rendering":
    (check if "currentPath" is bigger than just "/") && (do split and stuff)
      .map(....)
  }
</ul>
```
note, i recomend using <ul> - unordered list, 
no need to have <ol> here (ol == ordered list), 
we don't care of the order number, 
coz we do not display bullet points of numbers here.
ordered list is slower and harder for rendering in the browser, 
especially if there are many items in it.

Reply:
I did it and used span